src/DependencyInjection/Compiler/AddDependencyCallsCompilerPass.php

```diff
@@ -382,11 +395,19 @@ public function fixTemplates(ContainerBuilder $container, Definition $definition
 
         $definedTemplates = $overwrittenTemplates['view'] + $definedTemplates;
 
+        $templateRegistryId = $serviceId.'.template_registry';
+        $templateRegistryDefinition = $container
+            ->register($templateRegistryId, TemplateRegistry::class)
+            ->addTag('sonata.admin.template_registry')
+            ->setPublic(true); // Temporary fix until we can support service locators
+
         if ($container->getParameter('sonata.admin.configuration.templates') !== $definedTemplates) {
-            $definition->addMethodCall('setTemplates', [$definedTemplates]);
+            $templateRegistryDefinition->addArgument($definedTemplates);
         } else {
-            $definition->addMethodCall('setTemplates', ['%sonata.admin.configuration.templates%']);
+            $templateRegistryDefinition->addArgument('%sonata.admin.configuration.templates%');
```

Comment:
now that you touch this line, can you fix this other PR? it is on the same line

https://github.com/sonata-project/SonataAdminBundle/pull/4842

Can you check if with your code works and otherwise add what the other PR is suggesting?

Reply:
Because this shows up in the PR main page now: We decided not to do this right now.